Add tabs to the UI

RRickyH opened the pull request:
- Adds a side bar with tabs to the UI.
 - Modify git ignore to ignore build files (dist/ *.spec).
 - Change methods to be class and static methods for calendar APIs.
 - Support connecting and disconnecting calendar's through the settings tab.

AaroSnid committed:
Add on topic reminder GUI support and functionality

AaroSnid committed:
Convert database to JSON and add Pomodoro timer GUI

RRickyH committed:
Add tabs to the UI

 - Adds a side bar with tabs to the UI.
 - Modify git ignore to ignore build files (dist/ *.spec).
 - Change methods to be class and static methods for calendar APIs.
 - Support connecting and disconnecting calendar's through the settings
tab.

AaroSnid committed:
Modidying the outside file system for events and calendar

- Changed all tokens to .txt files
- Changed all program logs to .json

AaroSnid commented on src_files/bibb.py:
```diff
@@ -2,17 +2,128 @@
+)
 
+from src_files.styles import ACTIVE_COLOR, INACTIVE_COLOR
+from src_files.tabs import GoalTab, HomeTab, PomodoroTab, SettingsTab
```
The imports are failing as the "src_files." module cannot be found.

RRickyH replied:
This might be an issue with the way we're running this. I'll look into this more but for now the run_dev script works.

AaroSnid commented on src_files/calendars.py:
```diff
@@ -109,13 +145,29 @@ def get_credentials(self):
+            try:
+                account.authenticate(scopes=["basic", "calendar_all"])
+            except ValueError as exc:
+                raise ConnectionException("💥💥💣 Outlook Connection Failed!") from exc
```
Absolutely not. Remove the emojis

RRickyH replied:
Done.

AaroSnid commented on src_files/tabs/pomodoro.py:
I would like this file to contain the pomodoro widget contained in my other pull request, and for the associated refrences. I can merge into here if it makes it easier.

RRickyH replied:
Done.

AaroSnid reviewed (changes requested):
I am unable to run any of the code, as all references to src_files causes an import error, and I cannot remove them myself to test as:
1. I am not familiar enough with the module creation syntax
2. It should run without having to modify it

RRickyH committed:
Merge branch 'feature/gui' into feature/add-initial-tab-gui

RRickyH committed:
Make token files consistent

 - Stores google token credentials in Gtoken.txt.
 - Stores outlook token credentials in Otoken.txt.
 - Supports pomodoro timers.
 - Fetches calendar events for the home tab.
